src/test/java/com/rishabh/restframework/listeners/TestListener.java

```diff
@@ -36,30 +36,36 @@ public static void setTestMethodName(String name) {
 	public void onTestFailure(ITestResult result) {
 		log.error("The name of the testcase failed is : " + result.getName());
 		extentTest.get().fail(result.getThrowable());
+		extentTest.get().log(Status.FAIL, getTestMethodName() + " Test is failed " + result.getThrowable());
 
 	}
 
 	// When Test case get Skipped, this method is called.
 	@Override
 	public void onTestSkipped(ITestResult result) {
 		log.debug("The name of the testcase Skipped is : " + result.getName());
+		extentTest.get().log(Status.SKIP, result.getName() + " Test is Skipped" +  result.getThrowable());
 	}
 
 	// When Test case get Started, this method is called.
 	@Override
 	public void onTestStart(ITestResult result) {
 		log.info("The name of test case started is : " + result.getName());
-		setTestMethodName(result.getMethod().getMethodName());
+		if (result.getMethod().getDescription() == null)
+			setTestMethodName(result.getMethod().getMethodName());
+		else
+			setTestMethodName(result.getMethod().getDescription());
 		log.debug("Test Method name is : " + getTestMethodName());
 		test = extent.createTest(getTestMethodName());
+		test.log(Status.INFO, getTestMethodName() + " Test has Started");
 		extentTest.set(test);
 	}
 
 	// When Test case get passed, this method is called.
 	@Override
 	public void onTestSuccess(ITestResult result) {
 		log.info("The name of the testcase passed is : " + result.getName());
-		extentTest.get().log(Status.PASS, "Test Passed");
+		extentTest.get().log(Status.PASS, getTestMethodName() + " Test Passed Successfully");
```

Comment:
Reference it will give Reports like below : -

4:12:44 am | shouldAnswerWithTrue Test Passed Successfully
<img width="524" alt="Screenshot 2021-05-27 at 3 11 44 PM" src="https://user-images.githubusercontent.com/38694449/119804536-0e8ca100-befe-11eb-95bb-729a1f1834a2.png">





		extentTest.get().log(Status.PASS, getTestMethodName() + " Test Passed Successfully");